bin/single_target

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env ruby
+
+require "aws-sdk-elasticloadbalancingv2"
+
+TARGETS_FILE_PATH = '/tmp/deploy_tools_single_tgs'
+
+function = ARGV[0]
+$host_id = ARGV[1]
+
+fail "Please specify AWS_PROFILE" if ENV["AWS_PROFILE"].nil?
+fail "Usage: single_target <deregister || register> <host_id>" if (function.nil? or $host_id.nil?)
+
+REGION = "us-west-2".freeze
+class AWSClient
+  class << self
+    def elbv2
+      @elbv2 ||= Aws::ElasticLoadBalancingV2::Client.new(region: REGION)
+    end
+  end
+end
+
+def find_host_tgs ()
+  all_tgs = AWSClient.elbv2.describe_target_groups.target_groups
+  host_tgs = []
+
+  all_tgs.each do |tg|
+    targets = AWSClient.elbv2.describe_target_health(target_group_arn: tg.target_group_arn)
+                    .target_health_descriptions.map(&:target)
+    target_ports = targets.map(&:port).uniq
+
+    if targets.map(&:id).include?($host_id)
+      host_tgs.push ({
```

Comment:
weird data structure choice, wouldn't hash with key like arn/name be better?

Reply:
Not sure I understood your idea.
So, instead of current list of hashes structured like this:
```
[
  {
     target_group_arn: arn1,
     target_group_name: name1,
     target_current_port: port1
  },
  {
     target_group_arn: arn2,
     target_group_name: name2,
     target_current_port: port2
  }
]
```
you propose to have something like this:
```
[
  {
     name1: [arn1, port1]
  },
  {
     name2: [arn2, port2]
  }
]
```
i.e the same list of hashes but each hash has unique key with a list as a value?
It maybe looks more compact but AFAIU a code for retrieving data from that structure will be harder to read (and it will also require a loop-in-loop).

Reply:
Or you were saying about this structure? :thinking: 
```
{
   name1: [arn1, port1],
   name2: [arn2, port2]
}
```

I'll try that

Reply:
more like
```
{
name1: { arn: arn1, port: port1 },
name2: { arn: arn2, port: port2 },
}

Reply:
oook. changed